bin/vtvalidate.js

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env node
+
+"use strict";
+
+var exists = require('fs').existsSync;
+var fs = require('fs');
+var zlib = require('zlib');
+
+var usage = 'usage:';
+usage += '\n  node bin/vtvalidate.js <vector tile>';
+
+var input_tile = process.argv[2];
+if (!input_tile) {
+   console.log(usage);
+   process.exit(1);
+}
+
+if (!exists(input_tile)) {
+    console.log(input_tile + ' does not exist');
+    process.exit(1);
+}
+
+var validator = require('../');
+
+var buffer = fs.readFileSync(input_tile);
+
+if (buffer[0] === 0x1F && buffer[1] === 0x8B) {
```

Comment:
Realizing we should likely be using file-sniff here, but that file-sniff's detection checks a third magic number: https://github.com/mapbox/mapbox-file-sniff/blob/master/index.js#L79 (not sure why).

Noticing that this also should likely support zlib deflate in addition to gzip like we do in https://github.com/mapbox/gzip-hpp/blob/master/include/gzip/utils.hpp

Reply:
Great point. `mapbox-file-sniff` doesnt have a general `gzip` or `zlib` output, which would be tricky since sniff returns multiple possible filetypes under the gzipped umbrella. In the meantime, I'll add zlib checks to this PR and we can ponder how to expose this in perhaps a separate new module or something.